python/pyarrow/src/CMakeLists.txt

```diff
@@ -0,0 +1,445 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#
+# arrow_python
+#
+
+cmake_minimum_required(VERSION 3.5)
+
+# RPATH settings on macOS do not affect install_name.
+# https://cmake.org/cmake/help/latest/policy/CMP0068.html
+if(POLICY CMP0068)
+  cmake_policy(SET CMP0068 NEW)
+endif()
+
+#
+# Define
+# CMAKE_MODULE_PATH: location of cmake_modules in python
+#
+
+get_filename_component(PYARROW_SOURCE_DIR ${CMAKE_SOURCE_DIR} DIRECTORY)
+get_filename_component(PYTHON_SOURCE_DIR ${PYARROW_SOURCE_DIR} DIRECTORY)
+get_filename_component(ARROW_SOURCE_DIR ${PYTHON_SOURCE_DIR} DIRECTORY)
+set(ARROW_CPP_SOURCE_DIR "${ARROW_SOURCE_DIR}/cpp")
+
+# normalize ARROW_HOME path
+file(TO_CMAKE_PATH "$ENV{ARROW_HOME}" ARROW_HOME)
+set(CMAKE_MODULE_PATH "${PYTHON_SOURCE_DIR}/cmake_modules" "${ARROW_HOME}/lib/cmake/arrow")
+
+#
+# Arrow version
+#
+
+set(ARROW_PYTHON_VERSION "9.0.0-SNAPSHOT")
+string(REGEX MATCH "^[0-9]+\\.[0-9]+\\.[0-9]+" ARROW_PYTHON_BASE_VERSION "${ARROW_PYTHON_VERSION}")
+# Need to set to ARRROW_VERSION before finding Arrow package!
+project(arrow_python VERSION "${ARROW_PYTHON_BASE_VERSION}")
```

Comment:
pyarrow_cpp?

Reply:
I think it is suggested to also change the name of the final library: https://github.com/apache/arrow/pull/13311#discussion_r925231102

which is fine by me.

Reply:
Is that a safe change? (I don't know to what extent external packages depend on the exact library name?)

Reply:
Oh, didn't think about that.
It also implies a lot of changes in this PR alone and so I would actually be in favour of not changing the name of the final lib.

Reply:
Let's make it a separate JIRA where we can decide whether we want to do it or not.

Reply:
CMake's project name isn't related to the final library name. It's safe to change this without breaking compatibility.

I think that all `python/pyarrow/src/` are rewritten in Cython eventually. So `libarrow_python.so` will be gone. (I think that we don't need to rename it.)

Reply:
Created https://issues.apache.org/jira/browse/ARROW-17154 to change the cmake project name as a follow-up.

Reply:
> I think that all `python/pyarrow/src/` are rewritten in Cython eventually

I am not aware of any current plans to rewrite the C++ code in Cython (rather the opposite?) 

(there is a ticket to rewrite the C++ _tests_ in cython, to avoid having both pytest and ctest tests for testing the python package)

Reply:
> I think that all `python/pyarrow/src/` are rewritten in Cython eventually.

That's basically impossible. Cython does not allow you to write C++ classes, for example.

Reply:
> That's basically impossible. Cython does not allow you to write C++ classes, for example.

Oh...